x/mint/keeper/keeper.go

```diff
@@ -137,56 +143,99 @@ func (k Keeper) GetProportions(ctx sdk.Context, mintedCoin sdk.Coin, ratio sdk.D
+	communityPoolGrowthCoins := sdk.NewCoins(communityPoolGrowthProportion)
+	k.Logger(ctx).Info(fmt.Sprintf("\t\t\t...community growth: %d to %s", communityPoolGrowthProportion.Amount.Int64(), communityPoolGrowthAddress))
+	err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, communityPoolGrowthAddress, communityPoolGrowthCoins)
 	if err != nil {
 		return err
 	}
 
-	// Take the current balance of the developer rewards pool and remove it from the supply offset
-	// We re-introduce the new supply at the end, in order to avoid any rounding discrepancies.
-	// developerAccountBalance := k.bankKeeper.GetBalance(ctx, k.accountKeeper.GetModuleAddress(types.DeveloperVestingModuleAcctName), mintedCoin.Denom)
-	// k.bankKeeper.AddSupplyOffset(ctx, mintedCoin.Denom, developerAccountBalance.Amount)
-
-	// // Take the new balance of the developer rewards pool and add it back to the supply offset deduction
-	// developerAccountBalance = k.bankKeeper.GetBalance(ctx, k.accountKeeper.GetModuleAddress(types.DeveloperVestingModuleAcctName), mintedCoin.Denom)
-	// k.bankKeeper.AddSupplyOffset(ctx, mintedCoin.Denom, developerAccountBalance.Amount.Neg())
+	// allocate pool allocation ratio to security budget pool
+	communityPoolSecurityBudgetAddress := k.GetSubmoduleAddress(types.CommunitySecurityBudgetSubmoduleName)
+	communityPoolSecurityBudgetProportion := k.GetProportions(ctx, mintedCoin, proportions.CommunityPoolSecurityBudget)
+	communityPoolSecurityBudgetCoins := sdk.NewCoins(communityPoolSecurityBudgetProportion)
+	k.Logger(ctx).Info(fmt.Sprintf("\t\t\t...community security budget: %d to %s", communityPoolSecurityBudgetProportion.Amount.Int64(), communityPoolSecurityBudgetAddress))
+	err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, communityPoolSecurityBudgetAddress, communityPoolSecurityBudgetCoins)
+	if err != nil {
+		return err
+	}
 
-	// subtract from original provision to ensure no coins left over after the allocations
-	communityPoolCoins := sdk.NewCoins(mintedCoin).Sub(stakingIncentivesCoins).Sub(poolIncentivesCoins).Sub(participationRewardCoins)
-	err = k.distrKeeper.FundCommunityPool(ctx, communityPoolCoins, k.accountKeeper.GetModuleAddress(types.ModuleName))
+	// sweep any remaining tokens to the community growth pool (this should NEVER happen, barring rounding imprecision)
+	remainingCoins := sdk.NewCoins(mintedCoin).
+		Sub(stakingIncentivesCoins).
+		Sub(strategicReserveCoins).
+		Sub(communityPoolGrowthCoins).
+		Sub(communityPoolSecurityBudgetCoins)
+	err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, communityPoolGrowthAddress, remainingCoins)
 	if err != nil {
 		return err
 	}
 
-	// call an hook after the minting and distribution of new coins
-	k.hooks.AfterDistributeMintedCoin(ctx, mintedCoin)
+	// call a hook after the minting and distribution of new coins
+	// see osmosis' pool incentives hooks.go for an example
+	// k.hooks.AfterDistributeMintedCoin(ctx, mintedCoin)
```

Comment:
why remove AfterDistributeMintedCoin ?

Reply:
It's used to distribute tokens _from_ the pools to their functions e.g. osmosis uses this to allocate tokens in the pool-incentives module, see this good [description](https://github.com/osmosis-labs/osmosis/commit/7868e01c9e2a068e450d1fd619ce03d88c108d04#diff-fa48e1c32d174cf9e176f956fe7f2bf4783b60834cca1f0035c38905771274d1R42)